sycl/test/esimd/memory_properties.cpp

```diff
@@ -1453,3 +1455,113 @@ test_slm_gather_scatter(int byte_offset32) {
+  __ESIMD_NS::prefetch(ptrf);
+  __ESIMD_NS::prefetch(ptrf, byte_offset32);
+  __ESIMD_NS::prefetch(ptrf, byte_offset64);
+  __ESIMD_NS::prefetch(ptrf, mask_n1);
+  __ESIMD_NS::prefetch(ptrf, byte_offset32, mask_n1);
+  __ESIMD_NS::prefetch(ptrf, byte_offset64, mask_n1);
+  __ESIMD_NS::prefetch(ptrf, byte_offset32, mask_n1);
+  __ESIMD_NS::prefetch(ptrf, byte_offset64, mask_n1);
+
+  // CHECK-COUNT-8: call void @llvm.genx.lsc.prefetch.stateless.v1i1.v1i64(<1 x i1> {{[^)]+}}, i8 0, i8 2, i8 1, i16 1, i32 0, i8 3, i8 1, i8 2, i8 0, <1 x i64> {{[^)]+}}, i32 0)
+  __ESIMD_NS::prefetch(ptrf, byte_offset32, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, byte_offset64, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, mask_n1, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, byte_offset32, mask_n1, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, byte_offset64, mask_n1, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, byte_offset32, mask_n1, props_cache_load);
+  __ESIMD_NS::prefetch(ptrf, byte_offset64, mask_n1, props_cache_load);
+
+  // 4) prefetch(usm, ...): same as (1), (2) above, but with VS > 1.
+  // CHECK-COUNT-4: call void @llvm.genx.lsc.prefetch.stateless.v16i1.v16i64(<16 x i1> {{[^)]+}}, i8 0, i8 1, i8 2, i16 1, i32 0, i8 3, i8 2, i8 1, i8 0, <16 x i64> {{[^)]+}}, i32 0)
+  prefetch<float, 32, 2>(ptrf, ioffset_n16);
+  prefetch<float, 32, 2>(ptrf, ioffset_n16_view);
+
+  prefetch<float, 32, 2>(ptrf, loffset_n16);
+  prefetch<float, 32, 2>(ptrf, loffset_n16_view);
+
+  // CHECK-COUNT-4: call void @llvm.genx.lsc.prefetch.stateless.v16i1.v16i64(<16 x i1> {{[^)]+}}, i8 0, i8 2, i8 1, i16 1, i32 0, i8 3, i8 2, i8 1, i8 0, <16 x i64> {{[^)]+}}, i32 0)
+  prefetch<float, 32, 2>(ptrf, ioffset_n16, props_cache_load);
+  prefetch<float, 32, 2>(ptrf, ioffset_n16_view, props_cache_load);
+
+  prefetch<float, 32, 2>(ptrf, loffset_n16, props_cache_load);
+  prefetch<float, 32, 2>(ptrf, loffset_n16_view, props_cache_load);
+
+  // CHECK-COUNT-4: call void @llvm.genx.lsc.prefetch.stateless.v16i1.v16i64(<16 x i1> {{[^)]+}}, i8 0, i8 1, i8 2, i16 1, i32 0, i8 3, i8 2, i8 1, i8 0, <16 x i64> {{[^)]+}}, i32 0)
+  prefetch<float, 32, 2>(ptrf, ioffset_n16, mask_n16);
+  prefetch<float, 32, 2>(ptrf, ioffset_n16_view, mask_n16);
+
+  prefetch<float, 32, 2>(ptrf, loffset_n16, mask_n16);
+  prefetch<float, 32, 2>(ptrf, loffset_n16_view, mask_n16);
+
+  // CHECK-COUNT-4: call void @llvm.genx.lsc.prefetch.stateless.v16i1.v16i64(<16 x i1> {{[^)]+}}, i8 0, i8 2, i8 1, i16 1, i32 0, i8 3, i8 2, i8 1, i8 0, <16 x i64> {{[^)]+}}, i32 0)
+  prefetch<float, 32, 2>(ptrf, ioffset_n16, mask_n16, props_cache_load);
+  prefetch<float, 32, 2>(ptrf, ioffset_n16_view, mask_n16, props_cache_load);
+
+  prefetch<float, 32, 2>(ptrf, loffset_n16, mask_n16, props_cache_load);
+  prefetch<float, 32, 2>(ptrf, loffset_n16_view, mask_n16, props_cache_load);
```

Comment:
Please add some more test cases like `prefetch<float, 16>(ptr, 0)` and `prefetch<float, 16>(ptr, 0, 1)` trying to create a situation of ambiguous call error.